flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/data/RowDataProjection.java

```diff
@@ -254,4 +258,78 @@ public MapData getMap(int pos) {
   public RowData getRow(int pos, int numFields) {
     return (RowData) getValue(pos);
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (!(o instanceof RowDataProjection)) {
+      return false;
+    }
+
+    RowDataProjection that = (RowDataProjection) o;
+    return deepEquals(that);
```

Comment:
do we need to check `o == null`?
Inside `deepEquals`,  `other.getRowKind()` may fail if `other == null`

Reply:
NVM. When `o == null`, `(o instanceof RowDataProjection)` = false, at line 268, it will return false directly.